
Status: sent upstream

2004-09-23  Robert Millan  <robertmh@gnu.org>

	Changes for i386-k*bsd-gnu support.
	* config/i386/linux.h: Define REG_NAME to allow overriding of register
	names in sc structure.
	* config/i386/linux-unwind.h: Refer to sc->REG_NAME(reg) instead of
	sc->reg.

diff -ur gcc.old/gcc/config/i386/linux-unwind.h gcc/gcc/config/i386/linux-unwind.h
--- gcc.old/gcc/config/i386/linux-unwind.h	2004-09-08 02:17:13.000000000 +0200
+++ gcc/gcc/config/i386/linux-unwind.h	2004-09-23 11:44:24.000000000 +0200
@@ -135,28 +135,28 @@
   else
     return _URC_END_OF_STACK;
 
-  new_cfa = sc->esp;
+  new_cfa = sc->REG_NAME(esp);
   fs->cfa_how = CFA_REG_OFFSET;
   fs->cfa_reg = 4;
   fs->cfa_offset = new_cfa - (long) context->cfa;
 
   /* The SVR4 register numbering macros aren't usable in libgcc.  */
   fs->regs.reg[0].how = REG_SAVED_OFFSET;
-  fs->regs.reg[0].loc.offset = (long)&sc->eax - new_cfa;
+  fs->regs.reg[0].loc.offset = (long)&sc->REG_NAME(eax) - new_cfa;
   fs->regs.reg[3].how = REG_SAVED_OFFSET;
-  fs->regs.reg[3].loc.offset = (long)&sc->ebx - new_cfa;
+  fs->regs.reg[3].loc.offset = (long)&sc->REG_NAME(ebx) - new_cfa;
   fs->regs.reg[1].how = REG_SAVED_OFFSET;
-  fs->regs.reg[1].loc.offset = (long)&sc->ecx - new_cfa;
+  fs->regs.reg[1].loc.offset = (long)&sc->REG_NAME(ecx) - new_cfa;
   fs->regs.reg[2].how = REG_SAVED_OFFSET;
-  fs->regs.reg[2].loc.offset = (long)&sc->edx - new_cfa;
+  fs->regs.reg[2].loc.offset = (long)&sc->REG_NAME(edx) - new_cfa;
   fs->regs.reg[6].how = REG_SAVED_OFFSET;
-  fs->regs.reg[6].loc.offset = (long)&sc->esi - new_cfa;
+  fs->regs.reg[6].loc.offset = (long)&sc->REG_NAME(esi) - new_cfa;
   fs->regs.reg[7].how = REG_SAVED_OFFSET;
-  fs->regs.reg[7].loc.offset = (long)&sc->edi - new_cfa;
+  fs->regs.reg[7].loc.offset = (long)&sc->REG_NAME(edi) - new_cfa;
   fs->regs.reg[5].how = REG_SAVED_OFFSET;
-  fs->regs.reg[5].loc.offset = (long)&sc->ebp - new_cfa;
+  fs->regs.reg[5].loc.offset = (long)&sc->REG_NAME(ebp) - new_cfa;
   fs->regs.reg[8].how = REG_SAVED_OFFSET;
-  fs->regs.reg[8].loc.offset = (long)&sc->eip - new_cfa;
+  fs->regs.reg[8].loc.offset = (long)&sc->REG_NAME(eip) - new_cfa;
   fs->retaddr_column = 8;
   return _URC_NO_REASON;
 }
diff -ur gcc.old/gcc/config/i386/linux.h gcc/gcc/config/i386/linux.h
--- gcc.old/gcc/config/i386/linux.h	2004-09-08 07:58:29.000000000 +0200
+++ gcc/gcc/config/i386/linux.h	2004-09-23 11:44:24.000000000 +0200
@@ -182,3 +182,6 @@
 #define NEED_INDICATE_EXEC_STACK 1
 
 #define MD_UNWIND_SUPPORT "config/i386/linux-unwind.h"
+
+/* This macro is overriden in i386/k*bsd-gnu.h */
+#define REG_NAME(reg) reg
